### PR TITLE
Add identityRef & don't hardcode cloud name anymore

### PR DESCRIPTION
**Pull request by EmilienM**

### CODE REVIEW

File: api/hypershift/v1beta1/hostedcluster_types.go [hunk 1]
```diff
@@ -1826,12 +1826,10 @@ type AzurePlatformSpec struct {
...
-	//
+	// IdentityRef is a reference to a secret holding OpenStack credentials
+	// to be used when reconciling the hosted cluster.
 	// +kubebuilder:validation:Required
-	// +required
-	CloudsYamlSecret corev1.LocalObjectReference `json:"cloudsYamlSecret"`
+	IdentityRef OpenStackIdentityReference `json:"identityRef"`
 
 	// Name of the secret that contains the CA cert needed to connect to the OpenStack cloud.
 	// +optional
```
Comment (EmilienM) [inline api/hypershift/v1beta1/hostedcluster_types.go:1835]: what to do with this one? I haven't figured whether we want to keep it.

File: api/hypershift/v1beta1/hostedcluster_types.go [hunk 2]
```diff
@@ -1900,6 +1898,20 @@ type OpenStackPlatformSpec struct {
 	Tags []string `json:"tags,omitempty"`
 }
 
+// OpenStackIdentityReference is a reference to an infrastructure
```
Comment (EmilienM) [inline api/hypershift/v1beta1/hostedcluster_types.go:1901]: copy pasta from upstream CAPO

File: api/hypershift/v1beta1/hostedcluster_types.go (hunk 1 above)
Comment (EmilienM) [reply api/hypershift/v1beta1/hostedcluster_types.go:1835]: I guess we could get rid of it by checking in the secret whether we have a CA cert.

**Review by mdbooth — APPROVED** [review]
@MaysaMacedo any thoughts?

**Pull request merged**